test/e2e/framework/util.go

```diff
@@ -1359,14 +1359,14 @@ func podNotPending(c clientset.Interface, podName, namespace string) wait.Condit
 // to terminate or if the pod terminated with an unexpected reason.
 func waitForPodTerminatedInNamespace(c clientset.Interface, podName, reason, namespace string) error {
 	return WaitForPodCondition(c, namespace, podName, "terminated due to deadline exceeded", PodStartTimeout, func(pod *v1.Pod) (bool, error) {
-		if pod.Status.Phase == v1.PodFailed {
-			if pod.Status.Reason == reason {
+		if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+			if reason == "" ||
+				strings.ToLower(strings.TrimSpace(pod.Status.Reason)) == strings.ToLower(strings.TrimSpace(reason)) {
```

Comment:
I'll remove the TrimSpace and ToLower.

@msau42 The general steps are; 1) call api `Delete` to delete the pod, 2) check for Delete  error, if none 3) call `WaitForPodTerminated` which calls `waitForPodTerminatedInNamespace`.

You wrote _But this change would cause the test to pass even if Pod deletion failed_: I think you're referring to a successful call to api `Delete`, the caller passing in "" for the reason but the pod's Reason != "". Once the pod's Status is Success or Failed the loop will return ok (assuming passed-in reason is "").  Are you concerned that if the pod's Reason != "" that the pod won't be deleted (due to this Reason) but the wait loop will say it was deleted? This is a good point but when I searched _pkg_ I didn't see cases where a Reason stops a delete but I could have missed them. 
The alternate view is that if a pod's Reason != "" and the caller passes in reason="", that the loop continues until the api `Get` fails (`IsNotFound` is handled by callers), or 5m. **If** a pod can be fully deleted when it has a non-blank Reason (which I think happens) then the loop incorrectly waits 5m, returns an error, and the caller thinks the pod was not deleted when it was. However, in searching logs I don't recall seeing this scenario. What I've seen is the test thinking a pod was deleted (storage client pod) when, in fact, it was not fully deleted. Then later the namespace cannot be deleted due to a strangler pod.